service/dap/server.go

```diff
@@ -637,6 +645,8 @@ func (s *Server) onDisconnectRequest(request *dap.DisconnectRequest) {
 			s.log.Error(err)
 		}
 	}
+	s.stopNoDebugProcess()
```

Comment:
I know you safeguard against this being nil in the function, but I think the code was more clear when you had an else here. Either we have a debugger or a noDebug process.

Reply:
done.
Strictly speaking, s.debugger != nil doesn't necessary mean noDebug mode (s.debugger could be in uninitialized state).

Reply:
Since debugger is a pointer and pointers have nil default values in Go (unlike C, as far as I know), you must be referring to a case of exiting early before neither debugger nor process have been launched. The else could be else-if instead or in addition to doing the check in stopNoDebugProcess()